libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -36,22 +61,66 @@ def _format_messages(messages: List[BaseMessage]) -> Tuple[Optional[str], List[D
                 for m in messages
             ]
     """
-    system = None
-    formatted_messages = []
+    system: Optional[str] = None
+    formatted_messages: List[Dict] = []
     for i, message in enumerate(messages):
-        if not isinstance(message.content, str):
-            raise ValueError("Anthropic Messages API only supports text generation.")
         if message.type == "system":
             if i != 0:
                 raise ValueError("System message must be at beginning of message list.")
+            if not isinstance(message.content, str):
+                raise ValueError(
+                    "System message must be a string, "
+                    f"instead was: {type(message.content)}"
+                )
             system = message.content
+            continue
+
+        role = _message_type_lookups[message.type]
+        content: Union[str, List[Dict]]
+
+        if not isinstance(message.content, str):
+            # parse as dict
+            assert isinstance(
+                message.content, list
+            ), "Anthropic message content must be str or list of dicts"
+
+            # populate content
+            content = []
+            for item in message.content:
+                if isinstance(item, str):
+                    content.append(
+                        {
+                            "type": "text",
+                            "text": item,
+                        }
+                    )
+                elif isinstance(item, dict):
+                    if "type" not in item:
+                        raise ValueError("Dict content item must have a type key")
+                    if item["type"] == "image_url":
```

Comment:
should we support "image_url" or "image"? as an Anthropic user i might expect "image" support

Reply:
Supports both. `image_url` is the only one that triggers the conversion logic though.